rfcs/2024-03-22-20170-trace-data-model.md

```diff
@@ -0,0 +1,196 @@
+# RFC 20170 - 2024-03-22 - Establish an Internal Trace Model
+
+As of today there is an already existing Log and Metric event model, the purpose is to establish an
+internal model for describing a trace event.
+
+## Context
+
+The [Ingest OpenTelemetry Traces RFC] was accepted, but there was a condition that an [internal trace model] MUST be
+established. Meanwhile, there's also the [Accept Datadog Traces RFC] which was established, but
+that primarily only works well with the [`datadog_traces` sink]. There was also a previous RFC which
+was concerned with [validating schemas], but never came attempted to define an event schema.  This leaves this RFC
+to establish that data model.
+
+## Cross cutting concerns
+
+- This is part of the effort to support [sending OpenTelemetry traces], it would also support
+  ingesting OpenTelemetry. As well as to provide a path for a canonical model which traces can be
+  translated to/from ideally without loss of fidelity.
+
+## Scope
+
+### In scope
+
+- Define a Vector Trace event model schema.
+- Support for Links between `TraceEvent`s.
+- Defining a mapping between the Vector trace event model to the Datadog trace event (one direction).
+- Defining a mapping between the Vector trace event model and the OpenTelemetry trace event
+  (bi-directional).
+
+### Out of scope
+
+- Change the `TraceEvent` type internally from a re-typing of `LogEvent`.
```

Comment:
I don't see why this must be out of scope, if the desired data model requires a change.

Reply:
I'm also a bit confused on this one. It seems like the RFC here is proposing a strongly-typed representation, which I think makes sense, but means we'd also have an actual struct representing the trace event.

Reply:
Removed